### PR TITLE
Update TextInput API

### PR DESCRIPTION
**Pull request by JoelMarcey**
- Make the examples runnable (both copy/paste and with the web player)
- Add a bit more information in props where needed.

Test Plan:

http://localhost:8079/react-native/docs/textinput.html

screencapture-localhost-8079-react-native-docs-textinput-html-1466735362193-0

### CODE REVIEW

**Comment by facebook-github-bot-4** [comment]
By analyzing the blame information on this pull request, we identified @Bhullnatik and @pletcher to be potential reviewers.

**Comment by JoelMarcey** [comment]
Ref #8203

File: Libraries/Components/TextInput/TextInput.js [hunk 1]
```diff
@@ -71,10 +88,62 @@ type Event = Object;
...
- *    <TextInput {...props} />
- *  </View>
+ * ```ReactNativeWebPlayer
+ * import React, { Component } from 'react';
+ * import { AppRegistry, View, TextInput } from 'react-native';
+ *
+ * class UselessTextInput extends Component {
+ *   constructor(props) {
+ *     super(props);
+ *     this.state = {
```
Comment (caabernathy) [inline Libraries/Components/TextInput/TextInput.js]: Is this.state.text used anywhere in this component? It doesn't seem so. I'd get rid of the whole constructor if it isn't.

File: Libraries/Components/TextInput/TextInput.js [hunk 2]
```diff
@@ -209,28 +284,28 @@ const TextInput = React.createClass({
      */
     maxLength: PropTypes.number,
     /**
-     * Sets the number of lines for a TextInput. Use it with multiline set to
+     * Sets the number of lines for a `TextInput`. Use it with multiline set to
      * true to be able to fill the lines.
```
Comment (caabernathy) [inline Libraries/Components/TextInput/TextInput.js]: Nit: `true`

File: Libraries/Components/TextInput/TextInput.js [hunk 3]
```diff
@@ -267,32 +342,42 @@ const TextInput = React.createClass({
...
+     * maintaining selection information for a document.
+     *
+     * Some functionality that can be performed with this instance is:
+     *
+     * - `blur()`
+     * - `focus()`
+     * - `update()`
+     *
+     * > You can refernce `DocumentSelectionState` in
+     * > `vendor/document/selection/DocumentSelectionState.js`
```
Comment (caabernathy) [inline Libraries/Components/TextInput/TextInput.js]: Perhaps an actual link to the file in code?

**Comment by caabernathy** [comment]
Just one example code clarification, otherwise the rest are just nits.

**Comment by JoelMarcey** [comment]
@caabernathy Updated

**Comment by caabernathy** [comment]
LGTM!

**Comment by caabernathy** [comment]
@facebook-github-bot shipit

**Comment by facebook-github-bot-6** [comment]
Thanks for importing. If you are an FB employee go to [Phabricator](https://our.intern.facebook.com/intern/opensource/github/pull_request/639336876243165/int_phab) to review.

File: Libraries/Components/TextInput/TextInput.js [hunk 4]
```diff
@@ -267,32 +335,42 @@ const TextInput = React.createClass({
...
+     * An instance of `DocumentSelectionState`, this is some state that is responsible for
+     * maintaining selection information for a document.
+     *
+     * Some functionality that can be performed with this instance is:
+     *
+     * - `blur()`
+     * - `focus()`
+     * - `update()`
+     *
+     * > You can refernce `DocumentSelectionState` in
```
Comment (pletcher) [inline Libraries/Components/TextInput/TextInput.js]: Nit: reference 

(saw this in an email, wanted to comment -- hope it's not too late)

File: Libraries/Components/TextInput/TextInput.js [hunk 5]
```diff
@@ -316,28 +394,28 @@ const TextInput = React.createClass({
...
     /**
-     * If true, the text field will blur when submitted.
+     * If `true`, the text field will blur when submitted.
      * The default value is true for single-line fields and false for
-     * multiline fields. Note that for multiline fields, setting blurOnSubmit
-     * to true means that pressing return will blur the field and trigger the
-     * onSubmitEditing event instead of inserting a newline into the field.
+     * multiline fields. Note that for multiline fields, setting `blurOnSubmit`
+     * to `true` means that pressing return will blur the field and trigger the
+     * `onSubmitEditin`g event instead of inserting a newline into the field.
```
Comment (pletcher) [inline Libraries/Components/TextInput/TextInput.js]: `onSubmitEditing`

**Pull request closed without merging**